### ISSUE TITLE
Build tests with rake without running the test_fixture

### ISSUE THREAD

**Issue by tjora** [ThrowTheSwitch/Ceedling#83]
Hi,
We are currently using Ceedling with unity tests on real hardware. When running the test with rake test:all it works fine, we got the test running and collecting the tests by writing a wrapper and calling that wrapper from the :test_fixture: part of project.yml. The wrapper uploads the firmware to the hardware and collects the text output from the tests.

Is there a way to run the build of collected tests without running the :test_fixture: part of the project.yml?

When debugging tests in the debugger with single stepping and so on, I would like to build the .out-files automatically, but manually loading the into the target with our debugger software. How could I do that without making a custom :test_fixture: for this usecase?

Is there a way to select different :test_fixture: section in the project.yml-file based on environment variables or rake switches?

**Comment by mvandervoord** [comment]
I don't believe there is an option for doing this at the moment. I'm also not sure why someone would want to do such a thing. 

I'm coming from this perspective: http://www.throwtheswitch.org/build/which

If that article doesn't convince you otherwise, I'd love to hear the rationale behind this feature request. I'm not against you convincing me I am wrong. ;)

**Comment by tjora** [comment]
Hi Mark,
Thanks for the response!

I agree that you should simulate what you could (which we do), but some
tests access hardware accessories and I would like to test this interaction
with hardware. In strict terms they are not unit-tests, but integration
tests/hw tests can still benefit a lot from being run in a test framework
like Ceedling/Unity.

However, the argument is almost the same for simulators for embedded
targets which is advocated in the blog-post
http://www.throwtheswitch.org/build/which. The debug support provided by
commercial tools like the IAR embedded workbench is quite extensive, and
far better than basic printf-debugging. IAR will prompt the user for
reloading the binary (.out) whenever it detects a change in it. So if
ceedling could just build all .out-files without running them, debugging
with IAR both on target and on simulator would be quite smooth.

I could hack this by just providing a dummy/empty :test_fixture:, but then
I need to change my project.yml whenever I need to debug a failing test. I
would prefer to avoid that.

Sigve

**Comment by mvandervoord** [comment]
Cool. I think I see where you are going.

I think you've touched on what is probably the best option. How would you feel if Ceedling let you specify multiple test_fixtures. It would default to one of them, but then allow you to override it when you call ceedling by specifying the fixture you want by name (something like "fixture[none]"). 

I think this would fix the problem you've run into, plus possibly provide others with interesting situations with another option.

What do you think?

Mark

**Comment by wko27** [comment]
Hello!

The requested feature would be very useful.  For example, when debugging, it would be preferable if Ceedling could generate the executable and we debug it via e.g. lldb without actually running the executable.

Thanks,
Andrew

**Comment by mkarlesky** [comment]
Closing this out as it has been open for far too long. `ceedling test:build_only` has been available for quite some time.
